### ISSUE TITLE
Date and Time traits should have sensible defaults

### ISSUE THREAD

**Issue by corranwebster** [enthought/traits#667]
Currently the `Date` and `Time` traits try to call `date()` and `time()` respectively.  Since these are invalid calls, there is no meaningful default available.  The traits should have `date.today` and `time.now` as `factory` attributes, respectively.

**Comment by rkern** [comment]
Can you rephrase? I'm looking at the implementation, and I'm not sure what you are referring to. They are both just appropriately-parameterized `BaseInstance` traits. I don't see where they try to call `date()` or `time()`. They default to `None`, which I think is the best choice. I do not believe that they should default to moving targets like `date.today()` or `time.now()`.

**Comment by rkern** [comment]
Now, the fact that `Date` and `Time` as `BaseInstance` instances rather than proper `TraitType`s means that you can't pass in a new `factory`. You can't just do `Date(factory=datetime.date.today)` if you wanted that behavior. Making `Date` and `Time` proper `TraitType`s would solve that problem.

But I still recommend that you leave the defaults to be `None`.

**Comment by corranwebster** [comment]
Urg.  You're right.  I'm being dumb.  Ignore this.
